llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -5688,8 +5688,7 @@ static void printTypeName(Type *ty, raw_ostream &nameStream) {
   for (;;) {
     if (auto pointerTy = dyn_cast<PointerType>(ty)) {
       nameStream << "p" << pointerTy->getAddressSpace();
-      ty = pointerTy->getPointerElementType();
-      continue;
+      return;
```

Comment:
I’m sorry if I missed something in the discussion, but to get equivalent code with and without opaque pointers, can we do
```c++
if (pointerTy->isOpaque())
  return;
ty = pointerTy->getPointerElementType();
continue;
```
here?

That means we get functions with types in their names and signatures without opaque pointers.
And we get functions without types in both, names and signature, with enabled opaque pointers.

Reply:
We could do this, but we will end up with `getPointerElementType` which is deprecated and can be removed any day since LLVM 15 was released. My plan was to remove every `getPointerElementType` and later fix all issues found while enabling opaque pointers.

Reply:
I updated this PR with new way of searching for matching functions, but I can leave this change as a backup if tomorrow they will remove the deprecated functions and proceed with changes that you suggested.

Reply:
Thanks! Both is fine with me, let’s see what @nhaehnle thinks.